src/rmp/src/zero_slack_strategy.cpp

```diff
@@ -24,17 +24,26 @@
 
 namespace rmp {
 
-std::vector<sta::Vertex*> GetNegativeEndpoints(sta::dbSta* sta)
+std::vector<sta::Vertex*> GetNegativeEndpoints(sta::dbSta* sta,
+                                               rsz::Resizer* resizer)
 {
   std::vector<sta::Vertex*> result;
 
   sta::dbNetwork* network = sta->getDbNetwork();
   for (sta::Vertex* vertex : *sta->endpoints()) {
-    sta::PortDirection* direction = network->direction(vertex->pin());
+    sta::Pin* pin = vertex->pin();
+    sta::PortDirection* direction = network->direction(pin);
     if (!direction->isInput()) {
       continue;
     }
 
+    if (resizer != nullptr) {
+      if (resizer->dontTouch(pin) || resizer->dontTouch(network->net(pin))
```

Comment:
@maliberty should this functionality be moved to odb instead? Seems like purely database related functionality

Reply:
As it operating in terms of sta objects it would make more sense to move to dbSta.